global/scripts/golangci-lint/run.sh

```diff
@@ -4,29 +4,23 @@ if [ -z "$SCRIPTS_DIR" ]; then
   export SCRIPTS_DIR="$(echo $(dirname "$(realpath "$0")") | sed 's|\(.*pipelines\).*|\1|')"
 fi
 
-customJsonFile="custom.json"
-defaultJsonFile="$SCRIPTS_DIR/global/scripts/golangci-lint/.golangci.json"
+defaultConfigFile="$SCRIPTS_DIR/global/scripts/golangci-lint/.golangci.yml"
 
-if [ -f ".golangci.json" ]; then
-  disableDefault=$(jq -r '.linters.disable' $defaultJsonFile)
-  disableSpecific=$(jq -r '.linters.disable' .golangci.json)
-  mergedDisable=$(echo "$disableDefault $disableSpecific" | jq -s 'add | unique')
-  mergedLinters=$(jq -s '.[0].linters + .[1].linters | del(.disable)' $defaultJsonFile .golangci.json)
-  result=$(jq --argjson disable "$mergedDisable" --argjson linters "$mergedLinters" -n '$linters * {"disable": $disable}')
-  echo "{\"linters\": $result}" > $customJsonFile
+# Use the project's custom config file if it exists
+if [ -f ".golangci.yml" ]; then
+  configPath=".golangci.yml"
 else
-  cp "$defaultJsonFile" $customJsonFile
+  configPath="$defaultConfigFile"
```

Comment:
That's a good ideia to have a specific project file when we need more control. But, what if I just want to "ADD" to the default configuration in the majority of the cases? It'll be very annoying to copy the default configuration always. And also, the default can change, so I'll have to go into all my projects and change them with the default?! Could you please address this situation? Where I just want to add to the default?

Reply:
"Adding" into the config in this case will only allow you to disable more linters but not to enable any disabled linters. That's just the limitation of this merging approach.